src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -163,7 +184,75 @@ internal void PerformBackup()
         internal BackupConfigInfo GetBackupConfigInfo(string databaseName)
         {
             return this.backupUtilities.GetBackupConfigInfo(databaseName);
-        }   
-        
+        }
+
+        /// <summary>
+        /// Create a backup task for execution and cancellation
+        /// </summary>
+        /// <param name="sqlTask"></param>
+        /// <returns></returns>
+        internal async Task<TaskResult> BackupTask(SqlTask sqlTask)
+        {
+            sqlTask.AddMessage("In progress", SqlTaskStatus.InProgress, true);
+            Task<TaskResult> completedTask = await Task.WhenAny(PerformTask(), CancelTask(sqlTask));
+            sqlTask.AddMessage(completedTask.Result.TaskStatus == SqlTaskStatus.Failed ? completedTask.Result.ErrorMessage : "Completed",
+                               completedTask.Result.TaskStatus);
+            return completedTask.Result;
+        }
+
+        private async Task<TaskResult> PerformTask()
+        {
+            // create a task to perform backup
+            return await Task.Factory.StartNew(() =>
+            {
+                TaskResult result = new TaskResult();
+                try
+                {
+                    this.backupUtilities.PerformBackup();
+                    result.TaskStatus = SqlTaskStatus.Succeeded;
+                }
+                catch (Exception ex)
+                {
+                    result.TaskStatus = SqlTaskStatus.Failed;
+                    result.ErrorMessage = ex.Message;
+                    if (ex.InnerException != null)
+                    {
+                        result.ErrorMessage += System.Environment.NewLine + ex.InnerException.Message;
+                    }
+                }
+                return result;
+            });
+        }
+
+        private async Task<TaskResult> CancelTask(SqlTask sqlTask)
+        {
+            // create a task for backup cancellation request
+            return await Task.Factory.StartNew(() =>
+            {
+                TaskResult result = new TaskResult();
+                while (true)
```

Comment:
> while (true) [](start = 15, length = 13)

we should use cancellation tokens and wait for the token to be signaled.  We can't check-in a spin loop like this.